WordPress/gutenberg
# Accessibility improvements for Block Inserter

## Description


This PR adds a Close button to the Block Inserter that displays all the time for screen readers only. This PR also makes an adjustment to useDialog() call passing focusOnMount: null to prevent duplicate focus on Inserter open.

Bug fix: This PR will fix a bug I noticed on the Widgets screen where Close button on Inserter wasn't working at all. Small tweak to useCallback() call fixed this up.

## How has this been tested?




1. Login to WP Admin.
2. Go to Posts and open an existing post or create a new one.
3. Select the Toggle block inserter button.
4. Notice how there is a Close block inserter button but it is only visible to screen readers and shows up regardless of the viewport.
5. Also notice how focus is still placed in the search input.

## Screenshots 

## Types of changes





Bug fix

## Checklist:
